src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ComputableGraphStructure.java

```diff
@@ -0,0 +1,286 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.cachemanager;
+
+import org.broadinstitute.hellbender.utils.Utils;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * This class computes a number of useful structural properties for DAG specified by the
+ * immediate parents and immediate children of each node.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class ComputableGraphStructure {
+
+    private final Set<String> nodeKeysSet;
+    private final Set<String> nodeTagsSet;
+    private final Map<String, Set<String>> allTagsMap;
+    private final Map<String, Set<String>> immediateDescendentsMap;
+    private final Map<String, Set<String>> immediateParentsMap;
+    private final Map<String, Set<String>> allDescendentsMap;
+    private final Map<String, Set<String>> allParentsMap;
+    private final Map<String, Integer> depthsMap;
```

Comment:
What other order do you have in mind? the idea is to have a pre-computed path in the graph for each computable node such that following the path and caching intermediate evaluation will provide all of the necessary inputs for computing the next node until you reach the destination.